raiden/channel/netting_channel.py

```diff
@@ -785,14 +785,41 @@ def state_transition(self, state_change):
     def serialize(self):
         return ChannelSerialization(self)
 
+    def __eq__(self, other):
+        if isinstance(other, Channel):
+            return self.serialize() == other.serialize()
```

Comment:
This fails when the block_number of a channel is more recent than in the other, as a consequence `test_snapshotting` is failing. It compares a frozen version from the snapshot with the one from a running node, and the running node has an up-to-date block number.